File: app/src/main/java/za/org/grassroot2/util/ViewAnimation.java
package za.org.grassroot2.util;

import android.support.v4.view.ViewCompat;
import android.support.v4.view.ViewPropertyAnimatorListenerAdapter;
import android.view.View;

/**
 * Created by qbasso on 13.10.2017.
 */

public class ViewAnimation {

    public static void fadeOut(View v) {
        ViewCompat.animate(v).alpha(0).setListener(new ViewPropertyAnimatorListenerAdapter(){
            @Override
            public void onAnimationEnd(View view) {
                super.onAnimationEnd(view);
                view.setVisibility(View.GONE);
            }
        });
    }

    public static void fadeIn(View v) {
        v.setVisibility(View.VISIBLE);
        ViewCompat.setAlpha(v, 0);
        ViewCompat.animate(v).alpha(1);
    }

}

File: app/src/main/java/za/org/grassroot2/util/ViewAnimation.kt
package za.org.grassroot2.util

import android.support.v4.view.ViewCompat
import android.support.v4.view.ViewPropertyAnimatorListenerAdapter
import android.view.View

/**
 * Created by qbasso on 13.10.2017.
 */

object ViewAnimation {

    fun fadeOut(v: View) {
        ViewCompat.animate(v).alpha(0f).setListener(object : ViewPropertyAnimatorListenerAdapter() {
            override fun onAnimationEnd(view: View?) {
                super.onAnimationEnd(view)
                view!!.visibility = View.GONE
            }
        })
    }

    fun fadeIn(v: View) {
        v.visibility = View.VISIBLE
        v.alpha = 0f
        ViewCompat.animate(v).alpha(1f)
    }

}
